enigma-swing/src/main/java/cuchaz/enigma/gui/Gui.java

```diff
@@ -209,12 +214,56 @@ public void actionPerformed(ActionEvent e) {
 		this.retranslateUi();
 	}
 
+	public RightPanel getRightPanel() {
+		return this.rightPanel;
+	}
+
+	/**
+	 * Sets the right panel to the given panel.
+	 * @param id the id of the new panel
+	 * @param updateStateIfCurrent if the provided id is equal to the current id, this parameter determines whether to update the visibility of the panel
+	 */
+	public void setRightPanel(String id, boolean updateStateIfCurrent) {
+		if (id.equals(this.rightPanel.getId()) && updateStateIfCurrent) {
+			// only save divider location if hiding panel
+			if (this.rightPanel.isVisible()) {
+				UiConfig.setRightPanelDividerLocation(id, this.splitRight.getDividerLocation());
+			}
+
+			// swap visibility
+			this.rightPanel.setVisible(!this.rightPanel.isVisible());
+		} else {
```

Comment:
This else changes the right panel even if it's the current one when `updateStateIfCurrent` is false

Reply:
what do you mean by this?
the previous `if` block runs if the panel is equal to the current one, there's no way for the second block to be triggered unless the panel is being swapped

Reply:
There is an and condition, so when `updateStateIfCurrent` is false it doesn't run

Reply:
you're right; my mistake